receiver/otlpreceiver/internal/metrics/otlp.go

```diff
@@ -39,6 +42,18 @@ func (r *Receiver) Export(ctx context.Context, req pmetricotlp.ExportRequest) (p
 	ctx = r.obsrecv.StartMetricsOp(ctx)
 	err := r.nextConsumer.ConsumeMetrics(ctx, md)
 	r.obsrecv.EndMetricsOp(ctx, dataFormatProtobuf, dataPointCount, err)
+	// Use appropiate status codes for permanent/non-permanent errors
+	if err != nil {
+		s, ok := status.FromError(err)
```

Comment:
After looking at https://github.com/open-telemetry/opentelemetry-collector/pull/8038, we will need to reconsider our overall approach. `status.Status` doesn't have an `Unwrap` method, so `consumererror.IsPermanent` will always be false when called on errors returned from this function.

The way I see it, we can:
1. Translate gRPC status codes to HTTP status codes in the OTLP/HTTP receiver
2. Perform this logic in a gRPC-only wrapper function similar to what the OTLP/HTTP receiver has.

What do you think?

Reply:
That's exactly what I'm planning to do once I get this merged. And that's why I put https://github.com/open-telemetry/opentelemetry-collector/pull/8038 to draft mode.

Reply:
>That's exactly what I'm planning to do once I get this merged.

I'm sorry, I'm not clear on what you mean here. Are you saying you plan to update https://github.com/open-telemetry/opentelemetry-collector/pull/8038 to do gRPC->HTTP status code translation?

Reply:
yeah, that's correct.